### ISSUE TITLE
Fix broken specs

### ISSUE THREAD

**Issue by johnnyshields** [gmailgem/gmail#75]
_From @jgrevich on November 5, 2012 0:37_

Fixes a few broken specs and handles some cases where exceptions differ in Ruby 1.8/1.9.

Can you guys double check this to see if there are any errors or a better way to handle these changes?

_Copied from original issue: nu7hatch/gmail/pull/75_

**Comment by johnnyshields** [comment]
_From @webcracy on October 4, 2013 17:37_

:+1: 

* All specs passed on 1.8.7
* All specs passed on 1.9.2 
* All specs passed on 1.9.3, but first I had to apply the changes suggested in this commit: https://github.com/dcu/gmail/commit/8fc82e8956d64a77921ed1588713b83b3b867854 (this commit is part of pull request #66, which addresses several needed things but unfortunately IMO can't be merged as is)

**Comment by johnnyshields** [comment]
_From @jgrevich on February 14, 2014 19:9_

@nu7hatch @webcracy What are your thoughts on dropping 1.8 support and focusing on 2/2.1?

**Comment by johnnyshields** [comment]
_From @webcracy on April 21, 2014 14:35_

@jgrevich sorry I missed this question. I just mentioned my opinion on this by chance in https://github.com/nu7hatch/gmail/issues/104#issuecomment-40933391.

TL;DR:
* 0.4.1.X family could support 1.9.3 and older rubies with the current enhancements, pending the fix of imap_extensions (documented in #96)
* then a 0.5.X family could support Ruby 2 (2.1.1?) and drop support for older rubies
</blockquote>

**Comment by johnnyshields** [comment]
_From @jgrevich on April 21, 2014 17:14_

@webcracy  +1 to that version strategy.  Let's get moving.
